### PR TITLE
Slate resolveuid

### PR DESCRIPTION
**Pull request by tiberiuichim**

### CODE REVIEW

**Comment by tiberiuichim** [comment]
@jenkins-plone-org please run jobs

**Comment by tiberiuichim** [comment]
@jenkins-plone-org please run jobs

**Comment by tisto** [comment]
@jenkins-plone-org please run jobs

File: src/plone/restapi/tests/test_blocks_deserializer.py [hunk 1]
```diff
@@ -308,3 +308,82 @@ def test_deserialize_subblocks_transformers(self):
             "html"
         ]
         self.assertEqual(block, u"<div>This stays</div>")
+
+    def test_slate_internal_link_deserializer(self):
+        """test_internal_link_deserializer."""
```
Comment (tisto) [inline src/plone/restapi/tests/test_blocks_deserializer.py:313]: @tiberiuichim I'd remove the comment line here entirely.

File: src/plone/restapi/tests/test_blocks_deserializer.py [hunk 2]
```diff
@@ -308,3 +308,82 @@ def test_deserialize_subblocks_transformers(self):
...
+            },
+            "6b2be2e6-9857-4bcc-a21a-29c0449e1c68": {"@type": "title"},
+        }
+        res = self.deserialize(blocks=blocks)
+        value = res.blocks["2caef9e6-93ff-4edf-896f-8c16654a9923"]["value"]
+        link = value[0]["children"][1]["children"][1]
+        resolve_link = link["data"]["link"]["internal"]["internal_link"][0]["@id"]
+        self.assertTrue(resolve_link.startswith("../resolveuid/"))
+
+    def test_slate_simple_link_deserializer(self):
```
Comment (tisto) [inline src/plone/restapi/tests/test_blocks_deserializer.py:360]: @tiberiuichim can you elaborate what this test checks in comparison to test_slate_internal_link_deserializer. I don't see it at first sight.

File: src/plone/restapi/tests/test_blocks_serializer.py [hunk 3]
```diff
@@ -189,3 +192,95 @@ def test_serialize_subblocks_transformers(self):
...
+
+        value = res["2caef9e6-93ff-4edf-896f-8c16654a9923"]["value"]
+        link = value[0]["children"][1]["children"][1]
+        resolve_link = link["data"]["link"]["internal"]["internal_link"][0]["@id"]
+
+        self.assertTrue(resolve_link == self.portal.absolute_url() + "/doc1")
+
+    def test_simple_link_serializer(self):
+        doc_uid = IUUID(self.portal["doc1"])
+        resolve_uid_link = "../resolveuid/{}".format(doc_uid)
```
Comment (tisto) [inline src/plone/restapi/tests/test_blocks_serializer.py:251]: @tiberiuichim can you elaborate why we have two different kinds of links in Slate? Pls, excuse my ignorance on Slate internals here.

File: src/plone/restapi/tests/test_blocks_serializer.py [hunk 4]
```diff
@@ -189,3 +192,95 @@ def test_serialize_subblocks_transformers(self):
...
+        }
+        res = self.serialize(
+            context=self.portal["doc1"],
+            blocks=blocks,
+        )
+        value = res["abc"]["value"]
+        link = value[0]["children"][1]["data"]["url"]
+        self.assertTrue(link, self.portal.absolute_url() + "/doc1")
+
+    def test_bogus(self):
```
Comment (tisto) [inline src/plone/restapi/tests/test_blocks_serializer.py:283]: @tiberiuichim huh? What's that? :)

**Review by tisto — COMMENTED** [review]
Overall this looks good to me at first sight. I guess we need to write some Cypress tests on the Volto level to make sure this all works properly.

File: src/plone/restapi/tests/test_blocks_serializer.py (hunk 4 above)
Comment (tiberiuichim) [reply src/plone/restapi/tests/test_blocks_serializer.py:283]: hmm... not sure how that ended up there. I thought someone else added it :)

File: src/plone/restapi/tests/test_blocks_serializer.py (hunk 3 above)
Comment (tiberiuichim) [reply src/plone/restapi/tests/test_blocks_serializer.py:251]: "simple link" is the "new" link plugin that I've prepared, the one that uses the default Volto [AddLinkForm](https://github.com/plone/volto/blob/master/src/components/manage/AnchorPlugin/components/LinkButton/AddLinkForm.jsx)

The "internal link" is the current implementation that existing slate deployments use, kept for backwards compatibility.

Comment (tiberiuichim) [reply src/plone/restapi/tests/test_blocks_serializer.py:251]: @tisto please not that it needs this merged into Volto: https://github.com/plone/volto/pull/2421

File: src/plone/restapi/tests/test_blocks_deserializer.py (hunk 1 above)
Comment (tiberiuichim) [reply src/plone/restapi/tests/test_blocks_deserializer.py:313]: yeah... I copied from old code. I'll remove them

File: src/plone/restapi/tests/test_blocks_deserializer.py (hunk 2 above)
Comment (tiberiuichim) [reply src/plone/restapi/tests/test_blocks_deserializer.py:360]: There's now two ways of doing links with volto-slate: the "old" way, with the sidebar popup and the 3 tabs and the "new" way, which also uses a much simpler data model. The "internal link" is for the "old way", the "simple link" is for the new way.

**Comment by tiberiuichim** [comment]
@jenkins-plone-org please run jobs

**Review by sneridagh — APPROVED** [review]
lgtm!!

**Comment by tisto** [comment]
@tiberiuichim @sneridagh I'd like to test this on the Volto side, make sure we have the proper Cypress tests, then we can merge this PR and release it.

**Comment by tisto** [comment]
@tiberiuichim @sneridagh released with plone.restapi 8.1.0: https://pypi.org/project/plone.restapi/8.1.0/

**Pull request merged**